src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +674,105 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
 	return ret;
 }
 
-key_value_callback_t ModifierManager::_modifier_effect_callback(
-	ModifierValue& modifier, Modifier::modifier_type_t type, key_value_callback_t default_callback,
-	ModifierEffectValidator auto effect_validator
+bool ModifierManager::_add_flattened_modifier_cb(
+	ModifierValue& modifier_value,
+	std::string_view const& prefix, std::string_view const& key, ast::NodeCPtr const& value
```

Comment:
None of these should be `const&`, all of them are small enough that it's cheaper to pass by value.